src/_pytest/hookspec.py

```diff
@@ -582,6 +582,10 @@ def pytest_warning_captured(warning_message, when, item):
         in a future release.
 
         The item being executed if ``when`` is ``"runtest"``, otherwise ``None``.
+
+    :param tuple context:
```

Comment:
This is nice, but pytest kind already has this concept and it is called `location`, for example `pytest_runtest_logstart` receives a `location` parameter:

```python
def pytest_runtest_logstart(nodeid, location):
    """ signal the start of running a single test item.

    This hook will be called **before** :func:`pytest_runtest_setup`, :func:`pytest_runtest_call` and
    :func:`pytest_runtest_teardown` hooks.

    :param str nodeid: full id of the item
    :param location: a triple of ``(filename, linenum, testname)``
    """
```

I suggest we follow the same approach here, except you might call the last parameter `name` and use your description. 👍

Reply:
Ok makes sense to stay with that naming. I will change context to location. At the moment `context` consists of `(filename, function, linenumber)`. I would prefer to not name `function` to `name` as it can be confusing. I can change the order in the way of the above example!

Reply:
Done!